packages/apps/src/types/plugin/sender.ts

```diff
@@ -7,7 +7,7 @@ import { IPlugin } from './plugin';
 /**
  * a plugin that can send activities
  */
-export interface ISender extends IPlugin {
+export type ISender<TCustomEvents extends {} | undefined = undefined> = IPlugin<TCustomEvents> & {
```

Comment:
is there a way to keep this an interface? Ideally the rule of thumb for the repo has been if it gets extended it should be an interface, otherwise use types.

Reply:
This is a good rule of thumb, but with generics, this gives me problems:
![image](https://github.com/user-attachments/assets/695506e6-285b-40ab-8621-005151f53de4)

Supposedly it's because you can't extend interfaces with types that are unions (which makes sense i think). I can add a comment here?

Reply:
https://www.totaltypescript.com/concepts/an-interface-can-only-extend-an-object-type

Reply:
https://github.com/microsoft/teams.ts/pull/205#discussion_r2080985506

Decided to fix this via ^